app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -1097,6 +1097,15 @@ public CommCareSessionService getSession() throws SessionUnavailableException {
         }
     }
 
+
+    public UserKeyRecord getRecordForCurrentUser() {
+        try {
+            return getSession().getUserKeyRecord();
+        } catch (SessionUnavailableException e) {
+            return null;

```

Comment:
Definitely. Returning null when the state is more or less inconceivable is generally an antipattern, and is always harder to track down as a source of errors. The try/catch loop can be tedious for sure. If the context we use this in are SessionAwareCommCareActivity classes, we could extend their root methods (onCreate/onResume/wherever the source of calls are) to have wrappers which catch this exception generically and perform a login redirect, which would prevent the need to do a bunch of explicit handling, since we know what we want to do ~any time the session is unavailable.